### ISSUE TITLE
 require() of ES Module not supported on 5.0.0

### ISSUE THREAD

**Issue by ecker00** [chalk/chalk#527]
On 5.0.0 I get the following error.
```
Error: require() of ES Module /node_modules/chalk/source/index.js from /myscript.ts not supported.
Instead change the require of index.js in /myscript.ts to a dynamic import() which is available in all CommonJS modules.
```

`/myscript.ts`:
```ts
import chalk from 'chalk';
console.log(chalk.green`Test`));
```

This is my `.tsconfig`
```json
{
  "compilerOptions": {
    "lib": ["ES5", "ES6", "ESNext"],
    "target": "ESNext",
    "module": "commonjs",
    "moduleResolution": "node",
    "outDir": "./build",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "sourceMap": true,
    "noImplicitAny": true
  }
}
```

Downgrading to `4.1.2` got it working fine. 👍

**Comment by Qix-** [comment]
Yep this exact error is why I don't personally use Typescript. I can never get around it, and the TS folks don't seem interested in fixing whatever the underlying cause is - or at least, fixing the surrounding UX.

We're not doing anything strange here per my understanding.

Lastly, **please check the release notes**. Templates do not work with the latest version out of the box anymore.

**Comment by eugene1g** [comment]
This issue goes beyond Typescirpt. Somehow `npm` ends up using the version of `chalk`, which is now ESM, so NPM scripts break - 
```
/Users/xxx/server/local-ci/node_modules/npm/lib/utils/explain-dep.js:1
const chalk = require('chalk')
              ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/xxx/server/local-ci/node_modules/npm/node_modules/chalk/source/index.js from /Users/xxx/server/local-ci/node_modules/npm/lib/utils/explain-dep.js not supported.
Instead change the require of index.js in /Users/xxx/server/local-ci/node_modules/npm/lib/utils/explain-dep.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/xxx/server/local-ci/node_modules/npm/lib/utils/explain-dep.js:1:15)
    at Object.<anonymous> (/Users/xxx/server/local-ci/node_modules/npm/lib/utils/explain-eresolve.js:5:49)
    at Object.<anonymous> (/Users/xxx/server/local-ci/node_modules/npm/lib/utils/setup-log.js:10:21)
    at Object.<anonymous> (/Users/xxx/server/local-ci/node_modules/npm/lib/npm.js:61:18)
    at module.exports (/Users/xxx/server/local-ci/node_modules/npm/lib/cli.js:21:15)
    at Object.<anonymous> (/Users/xxx/server/local-ci/node_modules/npm/bin/npx-cli.js:125:1) {
  code: 'ERR_REQUIRE_ESM'
```

**Comment by Qix-** [comment]
`npm --version` please? Did you just install npm, by chance? This is concerning.

**Comment by eugene1g** [comment]
This issue is new - after 5.0 got published. 
I use `yarn@3.1.0`, and in many of my projects I end up with NPM@7.24.2 in `node_modules`. That version should install `chalk@4`, but somehow I end up with `chalk@npm` which now resolved to 5.0

```
server/node-server % yarn info -R npm
└─ npm@npm:7.24.2
   ├─ Version: 7.24.2
   │
   ├─ Exported Binaries
   │  ├─ npm
   │  └─ npx
   │
   └─ Dependencies
...
      ├─ chalk@npm:* → npm:5.0.0
      ├─ chownr@npm:* → npm:2.0.0
```

The issue well could be in how yarn resolved dependencies rather than npm/chalk

**Comment by Qix-** [comment]
Sounds like a yarn or npm bug. Nothing we can do about it here. Sorry.
